apps/asap-server/test/handlers/research-output/index-handler.test.ts

```diff
@@ -20,36 +22,379 @@ describe('Research Output index handler', () => {
+  });
+
+  describe('Should process all events, and not relay on the order', () => {
+    it('receives on correct order the events created and updated', async () => {
+      const roID = 'ro-1234';
+      const createEv = createEvent(roID);
+      const updateEv = updateEvent(roID);
+      const researchOutputResponse = {
+        ...getResearchOutputResponse(),
+        id: roID,
+      };
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce({
+        ...researchOutputResponse,
+        title: 'Original title',
+      });
+
+      await indexHandler(createEv);
+      expect(algoliaIndexMock.deleteObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.saveObject).toHaveBeenCalledWith({
+        ...researchOutputResponse,
+        title: 'Original title',
+        objectID: researchOutputResponse.id,
+      });
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce({
+        ...researchOutputResponse,
+        title: 'New title',
+      });
+
+      await indexHandler(updateEv);
+      expect(algoliaIndexMock.deleteObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.saveObject).toHaveBeenCalledTimes(2);
+      expect(algoliaIndexMock.saveObject).toHaveBeenLastCalledWith({
+        ...researchOutputResponse,
+        title: 'New title',
+        objectID: researchOutputResponse.id,
+      });
+    });
+
+    it('receives on reverse order the events created and updated', async () => {
+      const roID = 'ro-1234';
+      const createEv = createEvent(roID);
+      const updateEv = updateEvent(roID);
+      const researchOutputResponse = {
+        ...getResearchOutputResponse(),
+        id: roID,
+      };
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce({
+        ...researchOutputResponse,
+        title: 'Updated title',
+      });
+
+      await indexHandler(updateEv);
+      expect(algoliaIndexMock.deleteObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.saveObject).toHaveBeenLastCalledWith({
+        ...researchOutputResponse,
+        title: 'Updated title',
+        objectID: researchOutputResponse.id,
+      });
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce({
+        ...researchOutputResponse,
+        title: 'Old title',
+      });
+      await indexHandler(createEv);
+
+      expect(algoliaIndexMock.deleteObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.saveObject).toHaveBeenLastCalledWith({
+        ...researchOutputResponse,
+        title: 'Old title',
+        objectID: researchOutputResponse.id,
+      });
+    });
+    it('receives on correct order the events created and unpublished', async () => {
+      const roID = 'ro-1234';
+      const createEv = createEvent(roID);
+      const unpublishedEv = unpublishedEvent(roID);
+
+      const researchOutputResponse = {
+        ...getResearchOutputResponse(),
+        id: roID,
+      };
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce(
```

Comment:
ok so - we know that these functions called individually would do what you have in your test - but the challenge here is this:

when they created and then unpublish very quickly BOTH attempts to fetch the RO by ID return 404 - this is because by the time the 'create' event is received the status already changes in squidex. So your `fetchById` should throw twice with a 404

Reply:
this is still unresolved - and for a reason. Both handler invocations should, once again, encounter 404s